src/db/write.rs

```diff
@@ -159,8 +269,17 @@ impl<'a> Write<'a> {
             .flush(&mut self.dag_write)
             .await
             .map_err(FlushError)?;
-
-        let commit = match self.meta {
+        let mut index_metas = Vec::new();
+        for (_, index) in self.indexes.into_iter() {
```

Comment:
use `.values()`?

Reply:
This is weird but `self.indexes.into_iter()` iterates `(String, Index)` (by value) but `self.indexes.values().into_iter()` iterates `(&str, &Index)` (by reference).